Automattic/wp-calypso
# Start: Fix isNewUser logic

Fixes https://github.com/Automattic/wp-calypso/issues/95347

## Proposed Changes

Change the way in which the `isNewUser` variable is calculated in Start, when the onboarding flow is completed. That variable is then used to decide the value of `isNew7DUserSite`, used to decide if we need to fire the `calypso_new_user_site_creation` event.

With the changes, `isNewUser` will be based on `response.created_account` response value that we get when we create a new user, during the onboarding flow.

## Why are these changes being made?


`isNewUser` was calculated on the value of `dependencies.username`. [We can see](https://github.com/Automattic/wp-calypso/pull/95375/files#diff-a621c4ea73ecba2ef5f903f325105824ebd61dcc12da5f505d2e0031bc4a4fc9R931-R944) how username has value even [when](https://github.com/Automattic/wp-calypso/pull/95375/files#diff-a621c4ea73ecba2ef5f903f325105824ebd61dcc12da5f505d2e0031bc4a4fc9R895-R897) the signup type is social and `created_account` is false.

That's not correct.

This led to the wrong firing of the `calypso_new_user_site_creation` event, even for old social accounts. It is terrible during our Start/Stepper experiment, as `calypso_new_user_site_creation` events are many more in Start due to this error.

## Testing Instructions



* Run yarn start-mock-wordpress-com
* Go to /start and use Google with an already existing account
* Go through the onboarding flow and you should not see `calypso_new_user_site_creation` being triggered.
* Go through production to check if you can instead reproduce the issue.
